spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -377,11 +377,23 @@ private CopyOnWriteOperation(SparkCopyOnWriteScan scan, IsolationLevel isolation
       this.isolationLevel = isolationLevel;
     }
 
-    private List<DataFile> overwrittenFiles() {
+    private DataFileSet overwrittenFiles() {
       if (scan == null) {
-        return ImmutableList.of();
+        return DataFileSet.create();
       } else {
-        return scan.tasks().stream().map(FileScanTask::file).collect(Collectors.toList());
+        return scan.tasks().stream()
+            .map(FileScanTask::file)
+            .collect(Collectors.toCollection(DataFileSet::create));
+      }
+    }
+
+    private DeleteFileSet rewritableDeletes() {
+      if (scan == null) {
+        return DeleteFileSet.create();
+      } else {
+        return scan.tasks().stream()
+            .flatMap(task -> task.deletes().stream())
+            .collect(Collectors.toCollection(DeleteFileSet::create));
       }
```

Comment:
We won't have deletes in this case right (this is in the copy on write implementation)? Don't think this would be needed here.

Reply:
Ah nvm the scan tasks may need to be applying delete files if it was in MoR mode earlier, and if we're doing a copy on write which removes an existing file, we still need to make sure the orphan DVs are removed.

So in that case this is required